salt/client/__init__.py

```diff
@@ -1705,8 +1705,14 @@ def pub(self,
                 timeout,
                 **kwargs)
 
-        master_uri = 'tcp://' + salt.utils.zeromq.ip_bracket(self.opts['interface']) + \
-                     ':' + six.text_type(self.opts['ret_port'])
+        #Don't use brackets around interface if using tcp transport
+        if self.opts['transport'] == 'tcp':
```

Comment:
Can we put this and the other block that is similar into their own function instead of duplicating?